docs/html/development/getting-started.rst

```diff
@@ -8,7 +8,55 @@ This document is meant to get you setup to work on pip and to act as a guide and
 reference to the the development setup. If you face any issues during this
 process, please `open an issue`_ about it on the issue tracker.
 
-Development tools
+Running Developent Version
+==========================
+
+Just run `python src/pip` from cloned pip repo (https://github.com/pypa/pip):
+
+.. code-block:: console
+
+    $ python src/pip
+    
+That's it!
+
+You may want to skip to next chapter to read on how to run tests, or read on
+for details on how it works.
+
+Running `src/pip` will load all `pip` modules and
+`dependencies <https://github.com/pypa/pip/tree/master/src/pip/_vendor)>`_
+from the clone directory. This way they do not need to be installed on your
+system. The process of bundling dependencies with a project sources is called
+`vendoring`, but because there is no PEP for that, other Python won't work
+this way and still require `pip` for installation.
+
+If you want to practice `virtualenv` development, then `pip` inside can be
+relinked directly to development sources by installing it in
+`development mode`. `pip install -e .` does that. This needs to be done only
+once.
+
+Full setup for `virtualenv` in `devpip` subdir:
+
+.. code-block:: console
+
+    $ virtualenv .devpip
+    $ source .devpip/bin/activate
+    # ^^^ Linux only, on Windows run `.devpip/Scripts/activate.bat`
+    (.devpip)$ pip install -e .
+
+To avoid activating `virtualenv` every time to test `pip`, you can use full path
+to the executable to run development version:
+
+.. code-block:: console
+
+    $ .devpip/bin/pip --version
+    pip 18.1.dev0 from /home/techtonik/p/pip/src/pip (python 2.7)
```

Comment:
Minor point: It would be better to show examples using Python 3.x. Python 2 is not something we should be encouraging people to use.

Reply:
Yep. Will do. But I will have to use `python3` name, because `python` on Ubuntu 18.04 [still points](https://ubuntuforums.org/showthread.php?t=2390462&p=13797513#post13797513) to 2nd version.